CONTRIBUTING.md

```diff
@@ -23,3 +23,55 @@ use GitHub pull requests for this purpose. Consult [GitHub Help] for more
 information on using pull requests.
 
 [GitHub Help]: https://help.github.com/articles/about-pull-requests/
+
+## Building Dagger
+
+Dagger is built with [`bazel`](https://bazel.build).
+
+### Setting up Bazel
+
+* Go to [http://bazel.build](http://bazel.build), click on `GET BAZEL`, and install Bazel.
+* In *Android Studio*, go to `Preferences > Plugins`.
+  * search for `bazel` and install the plugin.
+  * If no result shows up, click on `Search in repositories`, search for `bazel` and install the plugin.
+  * If you still get no result, then the bazel plugin isn't available for your version of Android Studio.
+    * Go to [Integrating Bazel with IDEs](https://docs.bazel.build/versions/master/ide.html), click on `Android Studio plugin`.
+	* The `COMPATIBILITY` column indicates which versions of Android Studio have an available Bazel plugin.
+	* You can find the corresponding numbering scheme in the file names of [Android Studio downloads](https://developer.android.com/studio/#downloads). You may need to [download a Preview](https://developer.android.com/studio/preview/).
+
+### Building Dagger from the command line
+
+* Clone the Dagger repository:
+
+```
+git clone git@github.com:google/dagger.git dagger
+```
+
+*  Build the Dagger project:
+
+```
+bazel build
+```
+
+* You can install the Dagger libraries in your local maven repository with the `install-local-snapshot` script.
+* It will build the libraries and install them with a `LOCAL-SNAPSHOT` version.
+* If you get the following error: `ERROR: missing input file '@androidsdk//:build-tools/26.0.2/aapt'`, install the missing build tools version with sdkmanager.
+
+```
+./util/install-local-snapshot.sh
+```
+
+You can run all tests with `bazel test //...` and a single test with the path and class name, e.g. `bazel test ///javatests/dagger/internal/codegen:KeyFactoryTest`. Learn more about Bazel targets [here](https://docs.bazel.build/versions/master/build-ref.html).
+
+
+### Importing the Dagger project in Android Studio
+
+* In Android Studio, select `Import Bazel Project`.
+* Input the path to the Dagger project under `workspace`, click `Next`.
+* Select `Generate from BUILD file`, type `BUILD` in the `Build file` input, click `Next`.
+* In the `Project View` form, uncomment one of the `android_sdk_platform` lines. Pick one that you have installed, then click `Finish`.
+  * If you get an error on Bazel sync, `Cannot run program "bazel"`, then:
+    * In the command line, run `where bazel` and copy the output  (e.g. `/usr/local/bin/bazel`)
+	* In Android Studio, go to `Preferences > Bazel Settings` and replace `Bazel binary location` with what you just copied.
+* Syncing will take a long time (could be 10 minutes). You can sync at anytime by click on the weird red and black icon at the top of Android Studio.
```

Comment:
This will depend on the project *cough* POS

Reply:
This is a guide for the dagger project, and it took 10 minutes on my machine to sync the dagger project. It's machine dependent, but still.

Reply:
How about this:

"The first sync can take a long time. When build files are changed, you can run partial syncs (which should be faster) from the file menu."

People will then see the icon in the menu, and be able to build that mental link